iwyu_output.cc

```diff
@@ -456,9 +456,16 @@ string MungedForwardDeclareLineForTemplates(const TemplateDecl* decl) {
   // The template name is now the last word on the line.  Replace it
   // by its fully-qualified form.  Apparently rfind's endpos
   // argument is inclusive, so substract one to get past the end-space.
-  const string::size_type name = line.rfind(' ', endpos - 1);
+  const char kFinalSpecifier[] = " final ";
+  string::size_type final_pos = line.find(kFinalSpecifier);
+  if (final_pos != string::npos) {
+    line.replace(final_pos, sizeof(kFinalSpecifier), " ");
+  }
+  string::size_type name = line.rfind(" ", endpos - 1);
   CHECK_(name != string::npos && "Unexpected printable template-type");
-  return PrintForwardDeclare(decl, line.substr(0, name), GlobalFlags().cxx17ns);
```

Comment:
I think this was fine as-was (for some value of fine :)), let's not introduce the named temporary `template_name`.